File: core-service/src/main/java/com/sandjelkovic/dispatchd/content/trakt/dto/SeasonTrakt.java
package com.sandjelkovic.dispatchd.content.trakt.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.RequiredArgsConstructor;

import java.time.Instant;
import java.util.HashMap;
import java.util.Map;

@Data
@RequiredArgsConstructor
@AllArgsConstructor
public class SeasonTrakt {

	private String number;

	private Map<String, String> ids = new HashMap<>();

	private String overview;

	@JsonProperty("first_aired")
	private Instant firstAired;

	@JsonProperty("episode_count")
	private Integer episodeCount;

	@JsonProperty("aired_episodes")
	private Integer airedEpisodes;

}

File: content-service/src/main/kotlin/com/sandjelkovic/dispatchd/content/trakt/dto/SeasonTrakt.kt
package com.sandjelkovic.dispatchd.content.trakt.dto

import com.fasterxml.jackson.annotation.JsonProperty
import java.time.ZonedDateTime

/**
 * @author sandjelkovic
 * @date 28.1.18.
 */
data class SeasonTrakt(
        val number: String? = null,
        val ids: Map<String, String> = mapOf(),
        val overview: String? = null,
        @JsonProperty("first_aired")
        val firstAired: ZonedDateTime? = null,
        @JsonProperty("episode_count")
        val episodeCount: Int? = null,
        @JsonProperty("aired_episodes")
        val airedEpisodes: Int? = null
)
